<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Holdout Testing with SIAMCAT â€¢ SIAMCAT</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Holdout Testing with SIAMCAT">
<meta property="og:description" content="">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><script type="text/javascript">
  var _paq = _paq || [];
  _paq.push(['trackPageView']);
  _paq.push(['enableLinkTracking']);
  (function() {
    var u="//tr-denbi.embl.de/heimdall/";
    _paq.push(['setTrackerUrl', u+'p.php']);
    _paq.push(['setSiteId', '1']);
    var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
    g.type='text/javascript'; g.async=true; g.defer=true; g.src=u+'p.js'; s.parentNode.insertBefore(g,s);
  })();
</script>
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SIAMCAT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/SIAMCAT_vignette.html">Get Started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/SIAMCAT_vignette.html">Introduction to SIAMCAT</a>
    </li>
    <li>
      <a href="../articles/SIAMCAT_read-in.html">SIAMCAT input formats</a>
    </li>
    <li>
      <a href="../articles/SIAMCAT_holdout.html">Holdout testing with SIAMCAT</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Functions</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://twitter.com/siamcat_dev">
    <span class="fa fa-twitter"></span>
     
  </a>
</li>
<li>
  <a href="https://github.com/zellerlab/siamcat">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Holdout Testing with SIAMCAT</h1>
                        <h4 class="author">Jakob Wirbel, Konrad Zych, and Georg Zeller</h4>
            <address class="author_afil">
      EMBL Heidelberg<br><a class="author_email" href="mailto:#"></a><a href="mailto:georg.zeller@embl.de">georg.zeller@embl.de</a>
      </address>
                  
            <h4 class="date">Date last modified: 2018-09-24</h4>
      
      
      <div class="hidden name"><code>SIAMCAT_holdout.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>One of the functionalities of the <code>SIAMCAT</code> package is the training of statistical machine learning models on metagenomics data. In this vignette, we demonstrate how such a model can be built on one dataset and then be applied on another, similarly processed holdout dataset. This might be of interest when comparing data from two different studies on the same disease.</p>
<p>In this vignette, we look at two datasets from studies on colorectal cancer (CRC). The first study from <a href="http://europepmc.org/abstract/MED/25432777">Zeller et al.</a> investigated metagenomic markers for CRC in a population in France, while the second study from <a href="https://europepmc.org/abstract/MED/26408641">Yu et al.</a> used samples from China for the same goal. Both datasets were profiled with the same <a href="https://github.com/motu-tool/mOTUs_v2">taxonomic profiling tool</a>, yielding the same taxonomic identifiers, which is required for holdout testing.</p>
</div>
<div id="load-the-data" class="section level1">
<h1 class="hasAnchor">
<a href="#load-the-data" class="anchor"></a>Load the Data</h1>
<p>The datasets can be found on the web repository for public metagenomics datasets from the <a href="https://www.embl.de/research/units/scb/zeller/index.html">Zeller group</a>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(SIAMCAT)

<span class="co"># this is data from Zeller et al., Mol. Syst. Biol. 2014</span>
fn.feat.fr  &lt;-
<span class="st">    'https://embl.de/download/zeller/FR-CRC/FR-CRC-N141_tax-ab-specI.tsv'</span>
fn.meta.fr  &lt;-
<span class="st">    'https://embl.de/download/zeller/FR-CRC/FR-CRC-N141_metadata.tsv'</span>

<span class="co"># this is the external dataset from Yu et al., Gut 2017</span>
fn.feat.cn  &lt;-
<span class="st">    'https://embl.de/download/zeller/CN-CRC/CN-CRC-N128_tax-ab-specI.tsv'</span>
fn.meta.cn  &lt;-
<span class="st">    'https://embl.de/download/zeller/CN-CRC/CN-CRC-N128_metadata.tsv'</span></code></pre></div>
<p>First of all, we build a <code>SIAMCAT</code> object using the data from the French study in the same way that we have seen before in the main <code>SIAMCAT</code> vignette.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># features</span>
<span class="co"># be vary of the defaults in R!!!</span>
feat.fr  &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/read.table">read.table</a></span>(fn.feat.fr, <span class="dt">sep=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>, <span class="dt">quote=</span><span class="st">""</span>,
    <span class="dt">check.names =</span> <span class="ot">FALSE</span>, <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)
<span class="co"># the features are counts, but we want to work with relative abundances</span>
feat.fr.rel &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/prop.table">prop.table</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/matrix">as.matrix</a></span>(feat.fr), <span class="dv">2</span>)

<span class="co"># metadata</span>
meta.fr  &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/read.table">read.table</a></span>(fn.meta.fr, <span class="dt">sep=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>, <span class="dt">quote=</span><span class="st">""</span>,
    <span class="dt">check.names=</span><span class="ot">FALSE</span>, <span class="dt">stringsAsFactors=</span><span class="ot">FALSE</span>)

<span class="co"># create SIAMCAT object</span>
siamcat.fr &lt;-<span class="st"> </span><span class="kw"><a href="../reference/siamcat.html">siamcat</a></span>(<span class="dt">feat=</span>feat.fr.rel, <span class="dt">meta=</span>meta.fr,
    <span class="dt">label=</span><span class="st">'Group'</span>, <span class="dt">case=</span><span class="st">'CRC'</span>)</code></pre></div>
<pre><code>## + starting create.label</code></pre>
<pre><code>## Label used as case:
##    CRC
## Label used as control:
##    CTR</code></pre>
<pre><code>## + finished create.label.from.metadata in 0.016 s</code></pre>
<pre><code>## + starting validate.data</code></pre>
<pre><code>## +++ checking overlap between labels and features</code></pre>
<pre><code>## + Keeping labels of 141 sample(s).</code></pre>
<pre><code>## +++ checking sample number per class</code></pre>
<pre><code>## +++ checking overlap between samples and metadata</code></pre>
<pre><code>## + finished validate.data in 0.041 s</code></pre>
<p>We can load the data from the Chinese study in a similar way and also create a <code>SIAMCAT</code> object for the holdout dataset.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># features</span>
feat.cn  &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/read.table">read.table</a></span>(fn.feat.cn, <span class="dt">sep=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>, <span class="dt">quote=</span><span class="st">""</span>,
    <span class="dt">check.names =</span> <span class="ot">FALSE</span>)
feat.cn.rel &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/prop.table">prop.table</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/matrix">as.matrix</a></span>(feat.cn), <span class="dv">2</span>)

<span class="co"># metadata</span>
meta.cn  &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/read.table">read.table</a></span>(fn.meta.cn, <span class="dt">sep=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>, <span class="dt">quote=</span><span class="st">""</span>,
    <span class="dt">check.names=</span><span class="ot">FALSE</span>, <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)

<span class="co"># SIAMCAT object</span>
siamcat.cn &lt;-<span class="st"> </span><span class="kw"><a href="../reference/siamcat.html">siamcat</a></span>(<span class="dt">feat=</span>feat.cn.rel, <span class="dt">meta=</span>meta.cn,
        <span class="dt">label=</span><span class="st">'Group'</span>, <span class="dt">case=</span><span class="st">'CRC'</span>)</code></pre></div>
<pre><code>## + starting create.label</code></pre>
<pre><code>## Label used as case:
##    CRC
## Label used as control:
##    CTR</code></pre>
<pre><code>## + finished create.label.from.metadata in 0.001 s</code></pre>
<pre><code>## + starting validate.data</code></pre>
<pre><code>## +++ checking overlap between labels and features</code></pre>
<pre><code>## + Keeping labels of 128 sample(s).</code></pre>
<pre><code>## +++ checking sample number per class</code></pre>
<pre><code>## +++ checking overlap between samples and metadata</code></pre>
<pre><code>## + finished validate.data in 0.034 s</code></pre>
</div>
<div id="model-building-on-the-french-dataset" class="section level1">
<h1 class="hasAnchor">
<a href="#model-building-on-the-french-dataset" class="anchor"></a>Model Building on the French Dataset</h1>
<div id="preprocessing" class="section level2">
<h2 class="hasAnchor">
<a href="#preprocessing" class="anchor"></a>Preprocessing</h2>
<p>With the French dataset, we perform the complete process of model building within <code>SIAMCAT</code>, including data preprocessing steps like data validation, filtering, and data normalization.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">siamcat.fr &lt;-<span class="st"> </span><span class="kw"><a href="../reference/filter.features.html">filter.features</a></span>(
    siamcat.fr,
    <span class="dt">filter.method =</span> <span class="st">'abundance'</span>,
    <span class="dt">cutoff =</span> <span class="fl">0.001</span>,
    <span class="dt">rm.unmapped =</span> <span class="ot">TRUE</span>,
    <span class="dt">verbose=</span><span class="dv">2</span>
)</code></pre></div>
<pre><code>## + starting filter.features</code></pre>
<pre><code>## +++ before filtering, the data have 1754 features</code></pre>
<pre><code>## +++ removed 1 features corresponding to UNMAPPED reads</code></pre>
<pre><code>## +++ removed 1538 features whose values did not exceed 0.001 in any sample (retaining 215)</code></pre>
<pre><code>## + finished filter.features in 0.007 s</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">siamcat.fr &lt;-<span class="st"> </span><span class="kw"><a href="../reference/normalize.features.html">normalize.features</a></span>(
    siamcat.fr,
    <span class="dt">norm.method =</span> <span class="st">"log.std"</span>,
    <span class="dt">norm.param =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/list">list</a></span>(<span class="dt">log.n0 =</span> <span class="fl">1e-06</span>, <span class="dt">sd.min.q =</span> <span class="fl">0.1</span>),
    <span class="dt">verbose =</span> <span class="dv">2</span>
)</code></pre></div>
<pre><code>## + starting normalize.features</code></pre>
<pre><code>## +++ performing de novo normalization using the  log.std  method</code></pre>
<pre><code>## + feature sparsity before normalization: 46.05%</code></pre>
<pre><code>## +++ feature sparsity after normalization:      0 %</code></pre>
<pre><code>## + finished normalize.features in 0.004 s</code></pre>
</div>
<div id="model-training" class="section level2">
<h2 class="hasAnchor">
<a href="#model-training" class="anchor"></a>Model Training</h2>
<p>Now, we can build the statistical model. We use the same parameters as in the main <code>SIAMCAT</code> vignette, where the process is explained in more detail.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">siamcat.fr &lt;-<span class="st">  </span><span class="kw"><a href="../reference/create.data.split.html">create.data.split</a></span>(
    siamcat.fr,
    <span class="dt">num.folds =</span> <span class="dv">5</span>,
    <span class="dt">num.resample =</span> <span class="dv">2</span>
)</code></pre></div>
<pre><code>## Features splitted for cross-validation successfully.</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">siamcat.fr &lt;-<span class="st"> </span><span class="kw"><a href="../reference/train.model.html">train.model</a></span>(
    siamcat.fr,
    <span class="dt">method =</span> <span class="st">"lasso"</span>
)</code></pre></div>
<pre><code>## Trained lasso models successfully.</code></pre>
</div>
<div id="predictions" class="section level2">
<h2 class="hasAnchor">
<a href="#predictions" class="anchor"></a>Predictions</h2>
<p>Finally, we can make predictions for each cross-validation fold and evaluate the predictions as seen in the main <code>SIAMCAT</code> vignette.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">siamcat.fr &lt;-<span class="st"> </span><span class="kw"><a href="../reference/make.predictions.html">make.predictions</a></span>(siamcat.fr)</code></pre></div>
<pre><code>## Made predictions successfully.</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">siamcat.fr &lt;-<span class="st">  </span><span class="kw"><a href="../reference/evaluate.predictions.html">evaluate.predictions</a></span>(siamcat.fr)</code></pre></div>
<pre><code>## Evaluated predictions successfully.</code></pre>
</div>
</div>
<div id="application-on-the-holdout-dataset" class="section level1">
<h1 class="hasAnchor">
<a href="#application-on-the-holdout-dataset" class="anchor"></a>Application on the Holdout Dataset</h1>
<p>Now that we have successfully built the model for the French dataset, we can apply it to the Chinese holdout dataset. First, we will normalize the Chinese dataset with the same parameters that we used for the French dataset in order to make the data comparable. For that step, we can use the frozen normalization functionality in the <code>normalize.features</code> function in <code>SIAMCAT</code>. We supply to the function all normalization parameters saved in the <code>siamcat.fr</code> object, which can be accessed using the <code>norm_params</code> accessor.</p>
<div id="frozen-normalization" class="section level2">
<h2 class="hasAnchor">
<a href="#frozen-normalization" class="anchor"></a>Frozen Normalization</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">siamcat.cn &lt;-<span class="st"> </span><span class="kw"><a href="../reference/normalize.features.html">normalize.features</a></span>(siamcat.cn,
    <span class="dt">norm.param=</span><span class="kw"><a href="../reference/norm_params-methods.html">norm_params</a></span>(siamcat.fr),
    <span class="dt">feature.type=</span><span class="st">'original'</span>,
    <span class="dt">verbose =</span> <span class="dv">2</span>)</code></pre></div>
<pre><code>## + starting normalize.features</code></pre>
<pre><code>## + normalizing original features</code></pre>
<pre><code>## + performing frozen  log.std  normalization using the supplied parameters</code></pre>
<pre><code>## + feature sparsity before normalization: 49.77%</code></pre>
<pre><code>## + feature sparsity after normalization:     0%</code></pre>
<pre><code>## + finished normalize.features in 0.005 s</code></pre>
</div>
<div id="holdout-predictions" class="section level2">
<h2 class="hasAnchor">
<a href="#holdout-predictions" class="anchor"></a>Holdout Predictions</h2>
<p>Next, we apply the trained model to predict the holdout dataset.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">siamcat.cn &lt;-<span class="st"> </span><span class="kw"><a href="../reference/make.predictions.html">make.predictions</a></span>(
    <span class="dt">siamcat =</span> siamcat.fr,
    <span class="dt">siamcat.holdout =</span> siamcat.cn,
    <span class="dt">normalize.holdout =</span> <span class="ot">FALSE</span>)</code></pre></div>
<pre><code>## Warning in make.predictions(siamcat = siamcat.fr, siamcat.holdout =
## siamcat.cn, : WARNING: holdout set is not being normalized!</code></pre>
<pre><code>## Made predictions successfully.</code></pre>
<p>Note that the <code>make.predictions</code> function can also take care of the normalization of the holdout dataset.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## Alternative Code, not run here
siamcat.cn &lt;-<span class="st"> </span><span class="kw"><a href="../reference/siamcat.html">siamcat</a></span>(<span class="dt">feat=</span>feat.cn.rel, <span class="dt">meta=</span>meta.cn,
    <span class="dt">label=</span><span class="st">'Group'</span>, <span class="dt">case=</span><span class="st">'CRC'</span>)
siamcat.cn &lt;-<span class="st"> </span><span class="kw"><a href="../reference/make.predictions.html">make.predictions</a></span>(<span class="dt">siamcat =</span> siamcat.fr,
    <span class="dt">siamcat.holdout =</span> siamcat.cn,
    <span class="dt">normalize.holdout =</span> <span class="ot">TRUE</span>)</code></pre></div>
<p>Again, we have to evaluate the predictions:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">siamcat.cn &lt;-<span class="st"> </span><span class="kw"><a href="../reference/evaluate.predictions.html">evaluate.predictions</a></span>(siamcat.cn)</code></pre></div>
</div>
</div>
<div id="model-evaluation" class="section level1">
<h1 class="hasAnchor">
<a href="#model-evaluation" class="anchor"></a>Model Evaluation</h1>
<p>Now, we can compare the performance of the classifier on the original and the holdout dataset by using the <code>model.evaluation.plot</code> function. Here, we can supply several <code>SIAMCAT</code> objects for which the model evaluation will be plotted in the same plot. Note that we can supply the objects as named objects in order to print the names in the legend.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/model.evaluation.plot.html">model.evaluation.plot</a></span>(<span class="st">'FR-CRC'</span>=siamcat.fr,
    <span class="st">'CN-CRC'</span>=siamcat.cn,
    <span class="dt">colours=</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">'dimgrey'</span>, <span class="st">'orange'</span>))</code></pre></div>
<p><img src="SIAMCAT_holdout_files/figure-html/eval_plot_hidden-1.png" width="576" style="display: block; margin: auto auto auto 0;"></p>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session Info</h1>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/sessionInfo">sessionInfo</a></span>()</code></pre></div>
<pre><code>## R version 3.5.3 (2019-03-11)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS Mojave 10.14.3
## 
## Matrix products: default
## BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] SIAMCAT_1.2.0     phyloseq_1.24.2   mlr_2.13          ParamHelpers_1.12
## [5] BiocStyle_2.8.2  
## 
## loaded via a namespace (and not attached):
##  [1] nlme_3.1-137        matrixStats_0.54.0  fs_1.2.6           
##  [4] RColorBrewer_1.1-2  progress_1.2.0.9000 PRROC_1.3.1        
##  [7] rprojroot_1.3-2     tools_3.5.3         backports_1.1.3    
## [10] R6_2.4.0            vegan_2.5-2         lazyeval_0.2.1     
## [13] BiocGenerics_0.26.0 mgcv_1.8-27         colorspace_1.3-2   
## [16] permute_0.9-4       ade4_1.7-11         tidyselect_0.2.4   
## [19] gridExtra_2.3       prettyunits_1.0.2   compiler_3.5.3     
## [22] parallelMap_1.3     glmnet_2.0-16       Biobase_2.40.0     
## [25] xml2_1.2.0          desc_1.2.0          bookdown_0.7       
## [28] scales_1.0.0        checkmate_1.9.1     pkgdown_1.3.0      
## [31] commonmark_1.5      stringr_1.4.0       digest_0.6.18      
## [34] rmarkdown_1.10      XVector_0.20.0      pkgconfig_2.0.2    
## [37] htmltools_0.3.6     rlang_0.3.1         rstudioapi_0.7     
## [40] BBmisc_1.11         bindr_0.1.1         jsonlite_1.5       
## [43] dplyr_0.7.6         magrittr_1.5        biomformat_1.8.0   
## [46] Matrix_1.2-15       Rcpp_1.0.0          munsell_0.5.0      
## [49] S4Vectors_0.18.3    Rhdf5lib_1.2.1      ape_5.1            
## [52] infotheo_1.2.0      stringi_1.3.1       pROC_1.13.0        
## [55] yaml_2.2.0          MASS_7.3-51.1       zlibbioc_1.26.0    
## [58] rhdf5_2.24.0        plyr_1.8.4          grid_3.5.3         
## [61] LiblineaR_2.10-8    parallel_3.5.3      crayon_1.3.4       
## [64] lattice_0.20-38     Biostrings_2.48.0   splines_3.5.3      
## [67] multtest_2.36.0     hms_0.4.2           knitr_1.20         
## [70] beanplot_1.2        pillar_1.3.1        igraph_1.2.2       
## [73] reshape2_1.4.3      codetools_0.2-16    stats4_3.5.3       
## [76] fastmatch_1.1-0     XML_3.98-1.17       glue_1.3.0         
## [79] evaluate_0.11       data.table_1.12.0   foreach_1.4.4      
## [82] gtable_0.2.0        purrr_0.2.5         assertthat_0.2.0   
## [85] ggplot2_3.1.0       xfun_0.3            gridBase_0.4-7     
## [88] roxygen2_6.1.0      survival_2.43-3     tibble_2.0.1       
## [91] iterators_1.0.10    memoise_1.1.0       IRanges_2.14.10    
## [94] bindrcpp_0.2.2      cluster_2.0.7-1     corrplot_0.84</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#introduction">Introduction</a></li>
      <li><a href="#load-the-data">Load the Data</a></li>
      <li>
<a href="#model-building-on-the-french-dataset">Model Building on the French Dataset</a><ul class="nav nav-pills nav-stacked">
<li><a href="#preprocessing">Preprocessing</a></li>
      <li><a href="#model-training">Model Training</a></li>
      <li><a href="#predictions">Predictions</a></li>
      </ul>
</li>
      <li>
<a href="#application-on-the-holdout-dataset">Application on the Holdout Dataset</a><ul class="nav nav-pills nav-stacked">
<li><a href="#frozen-normalization">Frozen Normalization</a></li>
      <li><a href="#holdout-predictions">Holdout Predictions</a></li>
      </ul>
</li>
      <li><a href="#model-evaluation">Model Evaluation</a></li>
      <li><a href="#session-info">Session Info</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Konrad Zych, Jakob Wirbel, Georg Zeller.</p>
</div>

<div class="privacy">
  <p><strong>Privacy:</strong> Usage of this site follows
      <a href="https://www.embl.de/aboutus/privacy_policy/index.html">EMBLâ€™s Privacy Policy</a><br>
      In accordance with that policy, we use Matomo to collect anonymised data on visits to, downloads from, and searches of this site.
      Contact <a href="mailto:zeller@embl.de">Georg Zeller</a> for details.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
