<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>SIAMCAT input files formats • SIAMCAT</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.3.7/cosmo/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="SIAMCAT input files formats">
<meta property="og:description" content="">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><script type="text/javascript">
  var _paq = _paq || [];
  _paq.push(['trackPageView']);
  _paq.push(['enableLinkTracking']);
  (function() {
    var u="//tr-denbi.embl.de/heimdall/";
    _paq.push(['setTrackerUrl', u+'p.php']);
    _paq.push(['setSiteId', '1']);
    var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
    g.type='text/javascript'; g.async=true; g.defer=true; g.src=u+'p.js'; s.parentNode.insertBefore(g,s);
  })();
</script>
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SIAMCAT</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/SIAMCAT_vignette.html">Get Started</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/SIAMCAT_vignette.html">Introduction to SIAMCAT</a>
    </li>
    <li>
      <a href="../articles/SIAMCAT_read-in.html">SIAMCAT input formats</a>
    </li>
    <li>
      <a href="../articles/SIAMCAT_holdout.html">Holdout testing with SIAMCAT</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Functions</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://twitter.com/siamcat_dev">
    <span class="fa fa-twitter"></span>
     
  </a>
</li>
<li>
  <a href="https://github.com/zellerlab/siamcat">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>SIAMCAT input files formats</h1>
                        <h4 class="author">Konrad Zych, Jakob Wirbel, and Georg Zeller</h4>
            <address class="author_afil">
      EMBL Heidelberg<br><a class="author_email" href="mailto:#"></a><a href="mailto:georg.zeller@embl.de">georg.zeller@embl.de</a>
      </address>
                  
            <h4 class="date">Date last modified: 2018-09-24</h4>
      
      
      <div class="hidden name"><code>SIAMCAT_read-in.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>This vignette illustrates how to read and input your own data to the <code>SIAMCAT</code> package. We will cover reading in text files from the disk, formatting them and using them to create an object of <code>siamcat-class</code>.</p>
<p>The <code>siamcat-class</code> is the centerpiece of the package. All of the input data and result are stored inside of it. The structure of the object is described below in the <a href="#siamcat-class-object">siamcat-class object</a> section.</p>
</div>
<div id="loading-your-data-into-r" class="section level1">
<h1 class="hasAnchor">
<a href="#loading-your-data-into-r" class="anchor"></a>Loading your data into R</h1>
<div id="siamcat-input" class="section level2">
<h2 class="hasAnchor">
<a href="#siamcat-input" class="anchor"></a>SIAMCAT input</h2>
<p>Generally, there are three types of input for <code>SIAMCAT</code>:</p>
<div id="features" class="section level3">
<h3 class="hasAnchor">
<a href="#features" class="anchor"></a>Features</h3>
<p>The features should be a <code>matrix</code>, a <code>data.frame</code>, or an <code>otu_table</code>, organized as follows:</p>
<p><code>features (in rows) x samples (in columns).</code></p>
<table class="table">
<thead><tr class="header">
<th></th>
<th align="right">Sample_1</th>
<th align="right">Sample_2</th>
<th align="right">Sample_3</th>
<th align="right">Sample_4</th>
<th align="right">Sample_5</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><strong>Feature_1</strong></td>
<td align="right">0.59</td>
<td align="right">0.71</td>
<td align="right">0.78</td>
<td align="right">0.61</td>
<td align="right">0.66</td>
</tr>
<tr class="even">
<td><strong>Feature_2</strong></td>
<td align="right">0.00</td>
<td align="right">0.02</td>
<td align="right">0.00</td>
<td align="right">0.00</td>
<td align="right">0.00</td>
</tr>
<tr class="odd">
<td><strong>Feature_3</strong></td>
<td align="right">0.02</td>
<td align="right">0.00</td>
<td align="right">0.00</td>
<td align="right">0.00</td>
<td align="right">0.20</td>
</tr>
<tr class="even">
<td><strong>Feature_4</strong></td>
<td align="right">0.34</td>
<td align="right">0.00</td>
<td align="right">0.13</td>
<td align="right">0.07</td>
<td align="right">0.00</td>
</tr>
<tr class="odd">
<td><strong>Feature_5</strong></td>
<td align="right">0.06</td>
<td align="right">0.16</td>
<td align="right">0.00</td>
<td align="right">0.00</td>
<td align="right">0.00</td>
</tr>
</tbody>
</table>
<blockquote>
<p>Please note that <code>SIAMCAT</code> is supposed to work with <strong>relative abundances</strong>. Other types of data (e.g. counts) will also work, but not all functions of the package will result in meaningful outputs.</p>
</blockquote>
<p>An example of a typical feature file is attached to the <code>SIAMCAT</code> package, containing data from a publication investigating the microbiome in colorectal cancer (CRC) patients and controls (the study can be found here: <a href="http://europepmc.org/abstract/MED/25432777">Zeller et al</a>). The metagenomics data were processed with the <a href="http://mocat.embl.de/">MOCAT</a> pipeline, returning taxonomic profiles on the species levels (<code>specI</code>):</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(SIAMCAT)
fn.in.feat  &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(
    <span class="st">"extdata"</span>,
    <span class="st">"feat_crc_zeller_msb_mocat_specI.tsv"</span>,
    <span class="dt">package =</span> <span class="st">"SIAMCAT"</span>
)</code></pre></div>
<p>One way to load such data into <code>R</code> could be the use of <code>read.table</code></p>
<p><em>(Beware of the defaults in R! They are not always useful…)</em></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">feat &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/read.table">read.table</a></span>(fn.in.feat, <span class="dt">sep=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>,
    <span class="dt">header=</span><span class="ot">TRUE</span>, <span class="dt">quote=</span><span class="st">''</span>,
    <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>, <span class="dt">check.names =</span> <span class="ot">FALSE</span>)
<span class="co"># look at some features</span>
feat[<span class="dv">110</span><span class="op">:</span><span class="dv">114</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">2</span>]</code></pre></div>
<pre><code>##                                CCIS27304052ST-3-0 CCIS15794887ST-4-0
## Bacteroides caccae [h:1096]          1.557937e-03       1.761949e-03
## Bacteroides eggerthii [h:1097]       2.734527e-05       4.146882e-05
## Bacteroides stercoris [h:1098]       1.173786e-03       2.475838e-03
## Bacteroides clarus [h:1099]          4.830533e-04       4.589747e-06
## Methanohalophilus mahii [h:11]       0.000000e+00       0.000000e+00</code></pre>
</div>
<div id="metadata" class="section level3">
<h3 class="hasAnchor">
<a href="#metadata" class="anchor"></a>Metadata</h3>
<p>The metadata should be either a matrix or a <code>data.frame</code>.</p>
<p><code>samples (in rows) x metadata (in columns)</code>:</p>
<table class="table">
<thead><tr class="header">
<th></th>
<th align="right">Age</th>
<th align="right">Gender</th>
<th align="right">BMI</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><strong>Sample_1</strong></td>
<td align="right">52</td>
<td align="right">1</td>
<td align="right">20</td>
</tr>
<tr class="even">
<td><strong>Sample_2</strong></td>
<td align="right">37</td>
<td align="right">1</td>
<td align="right">18</td>
</tr>
<tr class="odd">
<td><strong>Sample_3</strong></td>
<td align="right">66</td>
<td align="right">2</td>
<td align="right">24</td>
</tr>
<tr class="even">
<td><strong>Sample_4</strong></td>
<td align="right">54</td>
<td align="right">2</td>
<td align="right">26</td>
</tr>
<tr class="odd">
<td><strong>Sample_5</strong></td>
<td align="right">65</td>
<td align="right">2</td>
<td align="right">30</td>
</tr>
</tbody>
</table>
<p>The <code>rownames</code> of the metadata should match the <code>colnames</code> of the feature matrix.</p>
<p>Again, an example of such a file is attached to the <code>SIAMCAT</code> package, taken from the same study:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fn.in.meta  &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(
    <span class="st">"extdata"</span>,
    <span class="st">"num_metadata_crc_zeller_msb_mocat_specI.tsv"</span>,
    <span class="dt">package =</span> <span class="st">"SIAMCAT"</span>
)</code></pre></div>
<p>Also here, the <code>read.table</code> can be used to load the data into <code>R</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">meta &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/read.table">read.table</a></span>(fn.in.meta, <span class="dt">sep=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>,
    <span class="dt">header=</span><span class="ot">TRUE</span>, <span class="dt">quote=</span><span class="st">''</span>,
    <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>, <span class="dt">check.names =</span> <span class="ot">FALSE</span>)
<span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/head">head</a></span>(meta)</code></pre></div>
<pre><code>##                    age gender bmi diagnosis localization crc_stage fobt
## CCIS27304052ST-3-0  52      1  20         0           NA         0    0
## CCIS15794887ST-4-0  37      1  18         0           NA         0    0
## CCIS74726977ST-3-0  66      2  24         1           NA         0    0
## CCIS16561622ST-4-0  54      2  26         0           NA         0    0
## CCIS79210440ST-3-0  65      2  30         0           NA         0    1
## CCIS82507866ST-3-0  57      2  24         0           NA         0    0
##                    wif_test
## CCIS27304052ST-3-0        0
## CCIS15794887ST-4-0        0
## CCIS74726977ST-3-0       NA
## CCIS16561622ST-4-0        0
## CCIS79210440ST-3-0        0
## CCIS82507866ST-3-0        0</code></pre>
</div>
<div id="label" class="section level3">
<h3 class="hasAnchor">
<a href="#label" class="anchor"></a>Label</h3>
<p>Finally, the label can come in different three different flavours:</p>
<ul>
<li>
<strong>Named vector</strong>: A named vector containing information about cases and controls. The names of the vector should match the <code>rownames</code> of the metadata and the <code>colnames</code> of the feature data. The label can contain either the information about cases and controls either
<ul>
<li>as integers (e.g. <code>0</code> and <code>1</code>),</li>
<li>as characters (e.g. <code>CTR</code> and <code>IBD</code>), or</li>
<li>as factors.</li>
</ul>
</li>
<li><p><strong>Metadata column</strong>: You can provide the name of a column in the metadata for the creation of the label. See below for an example.</p></li>
<li>
<strong>Label file</strong>: <code>SIAMCAT</code> has a function called <code>read.label</code>, which will create a label object from a label file. The file should be organized as follows:
<ul>
<li>The first line is supposed to read: <code>#BINARY:1=[label for cases];-1=[label for controls]</code>
</li>
<li>The second row should contain the sample identifiers as tab-separated list (consistent with feature and metadata).</li>
<li>The third row is then supposed to contain the actual class labels (tab-separated): <code>1</code> for each case and <code>-1</code> for each control.</li>
</ul>
<p>An example file is attached to the package again, if you want to have a look at it.</p>
</li>
</ul>
<p>For our example dataset, we can create the label object out of the metadata column called <code>diagnosis</code>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">label &lt;-<span class="st"> </span><span class="kw"><a href="../reference/create.label.html">create.label</a></span>(<span class="dt">meta=</span>meta, <span class="dt">label=</span><span class="st">"diagnosis"</span>,
    <span class="dt">case =</span> <span class="dv">1</span>, <span class="dt">control=</span><span class="dv">0</span>)</code></pre></div>
<p>When we later plot the results, it might be nicer to have names for the different groups stored in the label object (instead of <code>1</code> and <code>0</code>). We can also supply them to the <code>create.label</code> function:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">label &lt;-<span class="st"> </span><span class="kw"><a href="../reference/create.label.html">create.label</a></span>(<span class="dt">meta=</span>meta, <span class="dt">label=</span><span class="st">"diagnosis"</span>,
    <span class="dt">case =</span> <span class="dv">1</span>, <span class="dt">control=</span><span class="dv">0</span>,
    <span class="dt">p.lab =</span> <span class="st">'cancer'</span>, <span class="dt">n.lab =</span> <span class="st">'healthy'</span>)</code></pre></div>
<pre><code>## Label used as case:
##    1
## Label used as control:
##    0</code></pre>
<pre><code>## + finished create.label.from.metadata in 0.001 s</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">label<span class="op">$</span>info</code></pre></div>
<pre><code>## healthy  cancer 
##      -1       1</code></pre>
<blockquote>
<p>Note:<br>
If you have no label information for your dataset, you can still create a <code>SIAMCAT</code> object from your features alone. The <code>SIAMCAT</code> object without label information will contain a <code>TEST</code> label that can be used for making holdout predictions. Other functions, e.g. model training, will not work on such an object.</p>
</blockquote>
</div>
</div>
<div id="lefse-format-files" class="section level2">
<h2 class="hasAnchor">
<a href="#lefse-format-files" class="anchor"></a>LEfSe format files</h2>
<p><a href="https://bitbucket.org/biobakery/biobakery/wiki/lefse">LEfSe</a> is a tool for identification of associations between micriobial features and up to two metadata. LEfSe uses LDA (linear discriminant analysis).</p>
<p>LEfSe input file is a <code>.tsv</code> file. The first few rows contain the metadata. The following row contains sample names and the rest of the rows are occupied by features. The first column holds the row names:</p>
<table class="table">
<thead><tr class="header">
<th>label</th>
<th align="right">healthy</th>
<th align="right">healthy</th>
<th align="right">healthy</th>
<th align="right">cancer</th>
<th align="right">cancer</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><strong>age</strong></td>
<td align="right">52</td>
<td align="right">37</td>
<td align="right">66</td>
<td align="right">54</td>
<td align="right">65</td>
</tr>
<tr class="even">
<td><strong>gender</strong></td>
<td align="right">1</td>
<td align="right">1</td>
<td align="right">2</td>
<td align="right">2</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td><strong>Sample_info</strong></td>
<td align="right">Sample_1</td>
<td align="right">Sample_2</td>
<td align="right">Sample_3</td>
<td align="right">Sample_4</td>
<td align="right">Sample_5</td>
</tr>
<tr class="even">
<td><strong>Feature_1</strong></td>
<td align="right">0.59</td>
<td align="right">0.71</td>
<td align="right">0.78</td>
<td align="right">0.61</td>
<td align="right">0.66</td>
</tr>
<tr class="odd">
<td><strong>Feature_2</strong></td>
<td align="right">0.00</td>
<td align="right">0.02</td>
<td align="right">0.00</td>
<td align="right">0.00</td>
<td align="right">0.00</td>
</tr>
<tr class="even">
<td><strong>Feature_3</strong></td>
<td align="right">0.02</td>
<td align="right">0.00</td>
<td align="right">0.00</td>
<td align="right">0.00</td>
<td align="right">0.00</td>
</tr>
<tr class="odd">
<td><strong>Feature_4</strong></td>
<td align="right">0.34</td>
<td align="right">0.00</td>
<td align="right">0.43</td>
<td align="right">0.00</td>
<td align="right">0.00</td>
</tr>
<tr class="even">
<td><strong>Feature_5</strong></td>
<td align="right">0.56</td>
<td align="right">0.56</td>
<td align="right">0.00</td>
<td align="right">0.00</td>
<td align="right">0.00</td>
</tr>
</tbody>
</table>
<p>An example of such a file is attached to the <code>SIAMCAT</code> package:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fn.in.lefse&lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(
    <span class="st">"extdata"</span>,
    <span class="st">"LEfSe_crc_zeller_msb_mocat_specI.tsv"</span>,
    <span class="dt">package =</span> <span class="st">"SIAMCAT"</span>
)</code></pre></div>
<p><code>SIAMCAT</code> has a dedicated function to read LEfSe format files. The <code>read.lefse</code> function will read in the input file and extract metadata and features:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">meta.and.features &lt;-<span class="st"> </span><span class="kw"><a href="../reference/read.lefse.html">read.lefse</a></span>(fn.in.lefse,
    <span class="dt">rows.meta =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">6</span>, <span class="dt">row.samples =</span> <span class="dv">7</span>)
meta &lt;-<span class="st"> </span>meta.and.features<span class="op">$</span>meta
feat &lt;-<span class="st"> </span>meta.and.features<span class="op">$</span>feat</code></pre></div>
<p>We can then create a label object from one of the columns of the meta object and create a <code>siamcat</code> object:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">label &lt;-<span class="st"> </span><span class="kw"><a href="../reference/create.label.html">create.label</a></span>(<span class="dt">meta=</span>meta, <span class="dt">label=</span><span class="st">"label"</span>, <span class="dt">case =</span> <span class="st">"cancer"</span>)</code></pre></div>
<pre><code>## Label used as case:
##    cancer
## Label used as control:
##    healthy</code></pre>
<pre><code>## + finished create.label.from.metadata in 0.016 s</code></pre>
</div>
<div id="metagenomeseq-format-files" class="section level2">
<h2 class="hasAnchor">
<a href="#metagenomeseq-format-files" class="anchor"></a>metagenomeSeq format files</h2>
<p><a href="http://bioconductor.org/packages/metagenomeSeq/">metagenomeSeq</a> is an R package to determine differentially abundant features between multiple samples.</p>
<p>There are two ways to input data into metagenomeSeq:</p>
<ol style="list-style-type: lower-alpha">
<li>two files, one for metadata and one for features - those can be used in <code>SIAMCAT</code> just like described in <a href="#SIAMCAT-input">SIAMCAT input</a> with <code>read.table</code>:</li>
</ol>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fn.in.feat  &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/system.file">system.file</a></span>(
    <span class="st">"extdata"</span>,
    <span class="st">"CHK_NAME.otus.count.csv"</span>,
    <span class="dt">package =</span> <span class="st">"metagenomeSeq"</span>
)
feat &lt;-<span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/read.table">read.table</a></span>(fn.in.feat, <span class="dt">sep=</span><span class="st">'</span><span class="ch">\t</span><span class="st">'</span>,
    <span class="dt">header=</span><span class="ot">TRUE</span>, <span class="dt">quote=</span><span class="st">''</span>, <span class="dt">row.names =</span> <span class="dv">1</span>,
    <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>, <span class="dt">check.names =</span> <span class="ot">FALSE</span>
)</code></pre></div>
<ol start="2" style="list-style-type: lower-alpha">
<li>
<code>BIOM</code> format file, that can be used in <code>SIAMCAT</code> as described in the <a href="BIOM-format-files">following section</a>
</li>
</ol>
</div>
<div id="biom-format-files" class="section level2">
<h2 class="hasAnchor">
<a href="#biom-format-files" class="anchor"></a>BIOM format files</h2>
<p>The BIOM format files can be added to <code>SIAMCAT</code> via <code>phyloseq</code>. First the file should be imported using the <code>phyloseq</code> function <code>import_biom</code>. Then a <code>phyloseq</code> object can be imported as a <code>siamcat</code> object as descibed in the <a href="#Creating-a-siamcat-object-of-a-phyloseq-object">next section.</a></p>
</div>
<div id="creating-a-siamcat-object-of-a-phyloseq-object" class="section level2">
<h2 class="hasAnchor">
<a href="#creating-a-siamcat-object-of-a-phyloseq-object" class="anchor"></a>Creating a siamcat object of a phyloseq object</h2>
<p>The <code>siamcat</code> object extends on the <code>phyloseq</code> object. Therefore, creating a <code>siamcat</code> object from a <code>phyloseq</code> object is really straightforward. This can be done with the <code>siamcat</code> constructor function. First, however, we need to create a label object:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/data">data</a></span>(<span class="st">"GlobalPatterns"</span>) ## phyloseq example data
label &lt;-<span class="st"> </span><span class="kw"><a href="../reference/create.label.html">create.label</a></span>(<span class="dt">meta=</span><span class="kw">sample_data</span>(GlobalPatterns),
    <span class="dt">label =</span> <span class="st">"SampleType"</span>,
    <span class="dt">case =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"Freshwater"</span>, <span class="st">"Freshwater (creek)"</span>, <span class="st">"Ocean"</span>))</code></pre></div>
<pre><code>## Label used as case:
##    Freshwater,Freshwater (creek),Ocean
## Label used as control:
##    rest</code></pre>
<pre><code>## + finished create.label.from.metadata in 0.003 s</code></pre>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># run the constructor function</span>
siamcat &lt;-<span class="st"> </span><span class="kw"><a href="../reference/siamcat.html">siamcat</a></span>(<span class="dt">phyloseq=</span>GlobalPatterns, <span class="dt">label=</span>label)</code></pre></div>
<pre><code>## + starting validate.data</code></pre>
<pre><code>## +++ checking overlap between labels and features</code></pre>
<pre><code>## + Keeping labels of 26 sample(s).</code></pre>
<pre><code>## +++ checking sample number per class</code></pre>
<pre><code>## Data set has only 8 training examples of class Case .
## Note that a dataset this small/skewed is not necessarily suitable for analysis in this pipeline.</code></pre>
<pre><code>## +++ checking overlap between samples and metadata</code></pre>
<pre><code>## + finished validate.data in 0.443 s</code></pre>
</div>
</div>
<div id="creating-a-siamcat-class-object" class="section level1">
<h1 class="hasAnchor">
<a href="#creating-a-siamcat-class-object" class="anchor"></a>Creating a siamcat-class object</h1>
<p>The <code>siamcat-class</code> is the centerpiece of the package. All of the is stored inside of the object: <img src="siamcat.png" alt="internal make-up of a siamcat object"></p>
<p>In the figure above, rectangles depict slots of the object and the class of the object stored in the slot is given in the ovals. There are two obligatory slots -<strong>phyloseq</strong> (containing the metadata as <code>sample_data</code> and the original features as <code>otu_table</code>) and <strong>label</strong> - marked with thick borders.</p>
<p>The <code>siamcat</code> object is constructed using the <code><a href="../reference/siamcat.html">siamcat()</a></code> function. There are two ways to initialize it:</p>
<ul>
<li>
<p><strong>Features</strong>: You can provide a feature <code>matrix</code>, <code>data.frame</code>, or <code>otu_table</code> to the function (together with label and metadata information):</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">siamcat &lt;-<span class="st"> </span><span class="kw"><a href="../reference/siamcat.html">siamcat</a></span>(<span class="dt">feat=</span>feat, <span class="dt">label=</span>label, <span class="dt">meta=</span>meta)</code></pre></div>
</li>
<li>
<p><strong>phyloseq</strong>: The alternative is to create a <code>siamcat</code> object directly out of a <code>phyloseq</code> object:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">siamcat &lt;-<span class="st"> </span><span class="kw"><a href="../reference/siamcat.html">siamcat</a></span>(<span class="dt">phyloseq=</span>phyloseq, <span class="dt">label=</span>label)</code></pre></div>
</li>
</ul>
<p>Please note that you <strong>have to</strong> provide either <code>feat</code> or <code>phyloseq</code> and that you <strong>cannot</strong> provide both.</p>
<p>In order to explain the <code>siamcat</code> object better we will show how each of the slots is filled.</p>
<div id="phyloseq-label-and-orig_feat-slots" class="section level2">
<h2 class="hasAnchor">
<a href="#phyloseq-label-and-orig_feat-slots" class="anchor"></a>phyloseq, label and orig_feat slots</h2>
<p>The phyloseq and label slots are obligatory.</p>
<ul>
<li>The phyloseq slot is an object of class <code>phyloseq</code>, which is described in the help file of the <code>phyloseq</code> class. Help can be accessed by typing into R console: <code><a href="https://www.rdocumentation.org/packages/utils/topics/help">help('phyloseq-class')</a></code>.
<ul>
<li>The <code>otu_table</code> slot in <code>phyloseq</code> -see <code><a href="https://www.rdocumentation.org/packages/utils/topics/help">help('otu_table-class')</a></code>- stores the original feature table. For <code>SIAMCAT</code>, this slot can be accessed by <code>orig_feat</code>.</li>
</ul>
</li>
<li>The label slot contains a list. This list has a specific set of entries -see <code><a href="https://www.rdocumentation.org/packages/utils/topics/help">help('label-class')</a></code>- that are automatically generated by the <code>read.label</code> or <code>create.label</code> functions.</li>
</ul>
<p>The <code>phyloseq</code>, label and orig_feat are filled when the <code>siamcat</code> object is first created with the constructor function. <img src="allCr.png" alt="construction"></p>
</div>
<div id="all-the-other-slots" class="section level2">
<h2 class="hasAnchor">
<a href="#all-the-other-slots" class="anchor"></a>All the other slots</h2>
<p>Other slots are filled during the run of the <code>SIAMCAT</code> workflow: <img src="Slots_create.png" alt="workflow"></p>
</div>
<div id="accessing-and-assigning-slots" class="section level2">
<h2 class="hasAnchor">
<a href="#accessing-and-assigning-slots" class="anchor"></a>Accessing and assigning slots</h2>
<p>Each slot in <code>siamcat</code> can be accessed by typing</p>
<pre><code>slot_name(siamcat)</code></pre>
<p>e.g. for the <code>eval_data</code> slot you can types</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/eval_data-methods.html">eval_data</a></span>(siamcat)</code></pre></div>
<p>There is one notable exception: the phyloseq slot has to be accessed with <code><a href="../reference/physeq-methods.html">physeq(siamcat)</a></code> due to technical reasons.</p>
<p>Slots will be filled during the <code>SIAMCAT</code> workflow by the package’s functions. However, if for any reason a slot needs to be assigned outside of the workflow, the following formula can be used:</p>
<pre><code>slot_name(siamcat) &lt;- object_to_assign</code></pre>
<p>e.g. to assign a <code>new_label</code> object to the <code>label</code> slot:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/label-methods.html">label</a></span>(siamcat) &lt;-<span class="st"> </span>new_label</code></pre></div>
<p><em>Please note that this may lead to unforeseen consequences…</em></p>
</div>
<div id="slots-inside-the-slots" class="section level2">
<h2 class="hasAnchor">
<a href="#slots-inside-the-slots" class="anchor"></a>Slots inside the slots</h2>
<p>There are two slots that have slots inside of them. First, the <code>model_list</code> slot has a <code>models</code> slot that contains the actual list of <a href="https://mlr-org.github.io/mlr-tutorial/devel/html/index.html">mlr</a> models -can be accessed via <code><a href="../reference/models-methods.html">models(siamcat)</a></code>- and <code>model.type</code> which is a character with the name of the method used to train the model: <code><a href="../reference/model_type-methods.html">model_type(siamcat)</a></code>.</p>
<p>The phyloseq slot has a complex structure. However, unless the phyloseq object is created outside of the <code>SIAMCAT</code> workflow, only two slots of phyloseq slot will be occupied: the <code>otu_table</code> slot containing the features table and the <code>sam_data</code> slot containing metadata information. Both can be accessed by typing either <code>features(siamcat)</code> or <code><a href="../reference/meta-methods.html">meta(siamcat)</a></code>.</p>
<p>Additional slots inside the phyloseq slots do not have dedicated accessors, but can easily be reached once the phyloseq object is exported from the <code>siamcat</code> object:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">phyloseq &lt;-<span class="st"> </span><span class="kw"><a href="../reference/physeq-methods.html">physeq</a></span>(siamcat)
tax_tab &lt;-<span class="st"> </span><span class="kw">tax_table</span>(phyloseq)
<span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/head">head</a></span>(tax_tab)</code></pre></div>
<pre><code>## Taxonomy Table:     [6 taxa by 7 taxonomic ranks]:
##        Kingdom   Phylum          Class          Order         
## 549322 "Archaea" "Crenarchaeota" "Thermoprotei" NA            
## 522457 "Archaea" "Crenarchaeota" "Thermoprotei" NA            
## 951    "Archaea" "Crenarchaeota" "Thermoprotei" "Sulfolobales"
## 244423 "Archaea" "Crenarchaeota" "Sd-NA"        NA            
## 586076 "Archaea" "Crenarchaeota" "Sd-NA"        NA            
## 246140 "Archaea" "Crenarchaeota" "Sd-NA"        NA            
##        Family          Genus        Species                   
## 549322 NA              NA           NA                        
## 522457 NA              NA           NA                        
## 951    "Sulfolobaceae" "Sulfolobus" "Sulfolobusacidocaldarius"
## 244423 NA              NA           NA                        
## 586076 NA              NA           NA                        
## 246140 NA              NA           NA</code></pre>
<p>If you want to find out more about the phyloseq data structure, head over to the <a href="https://bioconductor.org/packages/release/bioc/html/phyloseq.html">phyloseq</a> BioConductor page. # Session Info</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/sessionInfo">sessionInfo</a></span>()</code></pre></div>
<pre><code>## R version 3.5.3 (2019-03-11)
## Platform: x86_64-apple-darwin15.6.0 (64-bit)
## Running under: macOS Mojave 10.14.3
## 
## Matrix products: default
## BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] SIAMCAT_1.2.0     phyloseq_1.24.2   mlr_2.13          ParamHelpers_1.12
## [5] BiocStyle_2.8.2  
## 
## loaded via a namespace (and not attached):
##  [1] nlme_3.1-137        matrixStats_0.54.0  fs_1.2.6           
##  [4] RColorBrewer_1.1-2  progress_1.2.0.9000 PRROC_1.3.1        
##  [7] rprojroot_1.3-2     tools_3.5.3         backports_1.1.3    
## [10] R6_2.4.0            vegan_2.5-2         lazyeval_0.2.1     
## [13] BiocGenerics_0.26.0 mgcv_1.8-27         colorspace_1.3-2   
## [16] permute_0.9-4       ade4_1.7-11         tidyselect_0.2.4   
## [19] gridExtra_2.3       prettyunits_1.0.2   compiler_3.5.3     
## [22] parallelMap_1.3     glmnet_2.0-16       Biobase_2.40.0     
## [25] xml2_1.2.0          desc_1.2.0          bookdown_0.7       
## [28] scales_1.0.0        checkmate_1.9.1     pkgdown_1.3.0      
## [31] commonmark_1.5      stringr_1.4.0       digest_0.6.18      
## [34] rmarkdown_1.10      XVector_0.20.0      pkgconfig_2.0.2    
## [37] htmltools_0.3.6     rlang_0.3.1         rstudioapi_0.7     
## [40] BBmisc_1.11         bindr_0.1.1         jsonlite_1.5       
## [43] dplyr_0.7.6         magrittr_1.5        biomformat_1.8.0   
## [46] Matrix_1.2-15       Rcpp_1.0.0          munsell_0.5.0      
## [49] S4Vectors_0.18.3    Rhdf5lib_1.2.1      ape_5.1            
## [52] infotheo_1.2.0      stringi_1.3.1       pROC_1.13.0        
## [55] yaml_2.2.0          MASS_7.3-51.1       zlibbioc_1.26.0    
## [58] rhdf5_2.24.0        plyr_1.8.4          grid_3.5.3         
## [61] LiblineaR_2.10-8    parallel_3.5.3      crayon_1.3.4       
## [64] lattice_0.20-38     Biostrings_2.48.0   splines_3.5.3      
## [67] multtest_2.36.0     hms_0.4.2           knitr_1.20         
## [70] beanplot_1.2        pillar_1.3.1        igraph_1.2.2       
## [73] reshape2_1.4.3      codetools_0.2-16    stats4_3.5.3       
## [76] fastmatch_1.1-0     glue_1.3.0          evaluate_0.11      
## [79] data.table_1.12.0   foreach_1.4.4       gtable_0.2.0       
## [82] purrr_0.2.5         assertthat_0.2.0    ggplot2_3.1.0      
## [85] xfun_0.3            gridBase_0.4-7      roxygen2_6.1.0     
## [88] survival_2.43-3     tibble_2.0.1        iterators_1.0.10   
## [91] memoise_1.1.0       IRanges_2.14.10     bindrcpp_0.2.2     
## [94] cluster_2.0.7-1     corrplot_0.84</code></pre>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#introduction">Introduction</a></li>
      <li>
<a href="#loading-your-data-into-r">Loading your data into R</a><ul class="nav nav-pills nav-stacked">
<li><a href="#siamcat-input">SIAMCAT input</a></li>
      <li><a href="#lefse-format-files">LEfSe format files</a></li>
      <li><a href="#metagenomeseq-format-files">metagenomeSeq format files</a></li>
      <li><a href="#biom-format-files">BIOM format files</a></li>
      <li><a href="#creating-a-siamcat-object-of-a-phyloseq-object">Creating a siamcat object of a phyloseq object</a></li>
      </ul>
</li>
      <li>
<a href="#creating-a-siamcat-class-object">Creating a siamcat-class object</a><ul class="nav nav-pills nav-stacked">
<li><a href="#phyloseq-label-and-orig_feat-slots">phyloseq, label and orig_feat slots</a></li>
      <li><a href="#all-the-other-slots">All the other slots</a></li>
      <li><a href="#accessing-and-assigning-slots">Accessing and assigning slots</a></li>
      <li><a href="#slots-inside-the-slots">Slots inside the slots</a></li>
      </ul>
</li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Konrad Zych, Jakob Wirbel, Georg Zeller.</p>
</div>

<div class="privacy">
  <p><strong>Privacy:</strong> Usage of this site follows
      <a href="https://www.embl.de/aboutus/privacy_policy/index.html">EMBL’s Privacy Policy</a><br>
      In accordance with that policy, we use Matomo to collect anonymised data on visits to, downloads from, and searches of this site.
      Contact <a href="mailto:zeller@embl.de">Georg Zeller</a> for details.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
